File: Php/core/Validation/CompositeValidator.php
<?php
namespace core\Validation{

	class CompositeValidator extends BaseValidator{

		/** @var Validator[] */
		private $validators;

		private $valid;
		private $validated;

		public function __construct(){
			$this->validators = array();
			$this->validated = false;
			$this->valid = true;
		}

		public function addValidator(Validator $validator){
			$this->validators[] = $validator;
		}

		public function validate(){
			$this->validated = true;
			foreach($this->validators as $validator){
				if(!$validator->isValid()){
					$this->valid = false;
				}
			}
		}

		public function isValid(){
			if(!$this->validated){
				$this->validate();
			}
			$this->ensureErrorArray();
			return $this->valid && count($this->errors) == 0;
		}

		public function getErrors(){
			$this->ensureErrorArray();
			foreach($this->validators as $validator){
				$this->errors = array_merge($this->errors, $validator->getErrors());
			}
			return $this->errors;
		}

	}

}

File: Web/Validators/CompositeValidator.cs
using System.Collections.Generic;

namespace Web.Validators{

	class CompositeValidator : BaseValidator{

		private List<IValidator> _validators;

		private bool _valid;
		private bool _validated;

	    public CompositeValidator()
	    {
	        _validators = new List<IValidator>();
			_validated = false;
			_valid = true;
	    }

		public void AddValidator(IValidator validator){
			_validators.Add(validator);
		}

	    protected override void Validate(){
			_validated = true;
			foreach(var validator in _validators){
				if(!validator.IsValid){
					_valid = false;
				}
			}
		}

	    public override bool IsValid
	    {
	        get
	        {
	            if (!_validated)
	            {
	                Validate();
	            }
	            EnsureErrorArray();
	            return _valid && Errors.Count == 0;
	        }
	    }

	    public override List<string> GetErrors(){
			EnsureErrorArray();
			foreach(var validator in _validators){
                Errors.AddRange(validator.GetErrors());
			}
			return Errors;
		}

	}

}